src/vs/workbench/contrib/notebook/browser/controller/executeActions.ts

```diff
@@ -82,17 +83,15 @@ async function runCell(editorGroupsService: IEditorGroupsService, context: INote
 	if (context.ui && context.cell) {
 		await context.notebookEditor.executeNotebookCells(Iterable.single(context.cell));
 		if (context.autoReveal) {
-			const cellIndex = context.notebookEditor.getCellIndex(context.cell);
-			context.notebookEditor.revealCellRangeInView({ start: cellIndex, end: cellIndex + 1 });
+			handleAutoReveal(context.cell, context.notebookEditor, configurationService);
 		}
 	} else if (context.selectedCells?.length || context.cell) {
 		const selectedCells = context.selectedCells?.length ? context.selectedCells : [context.cell!];
 		await context.notebookEditor.executeNotebookCells(selectedCells);
 		const firstCell = selectedCells[0];
 
 		if (firstCell && context.autoReveal) {
-			const cellIndex = context.notebookEditor.getCellIndex(firstCell);
-			context.notebookEditor.revealCellRangeInView({ start: cellIndex, end: cellIndex + 1 });
+			handleAutoReveal(firstCell, context.notebookEditor, configurationService);
```

Comment:
@Yoyokrazy I found it behaved differently from yesterday while running locally, and realized that it reveals after the cell is executed, which is already too late. for auto reveal (maybe when the private settings are enabled only), the cells should be revealed first before execution.